src/main/java/com/example/service/impl/BookServiceImpl.java

```diff
@@ -63,4 +64,10 @@ public List<BookDto> search(BookSearchParameters params, Pageable pageable) {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
```

Comment:
Do we need to setCategories during updating and saving book?

Reply:
Yes. Now it's working.

Reply:
Didn't find these logic.

Reply:
I implemented it in BookMapper interface. with help two methods:
@AfterMapping
    default void setCategoryIds(@MappingTarget BookDto bookDto, Book book) {
        bookDto.setCategoryIds(book.getCategories().stream()
                .map(Category::getId)
                .collect(Collectors.toSet()));
    }

    @AfterMapping
    default void setCategoryIds(@MappingTarget Book book, CreateBookRequestDto requestDto) {
        book.setCategories(requestDto.getCategoryIds().stream()
                .map(Category::new)
                .collect(Collectors.toSet()));
    }